var documenterSearchIndex = {"docs":
[{"location":"primer/","page":"Primer Design","title":"Primer Design","text":"CurrentModule = SimpleBio\nDocTestSetup = quote\n    using SimpleBio\nend","category":"page"},{"location":"primer/#Primer-Design","page":"Primer Design","title":"Primer Design","text":"","category":"section"},{"location":"primer/","page":"Primer Design","title":"Primer Design","text":"These are useful tools for primer design for PCR!","category":"page"},{"location":"primer/#Calculate-melting-temperature","page":"Primer Design","title":"Calculate melting temperature","text":"","category":"section"},{"location":"primer/","page":"Primer Design","title":"Primer Design","text":"dnatm\nstep_gen\nstep_from_end\nmultidnatm\nfwd_primers\nrev_primers","category":"page"},{"location":"primer/#SimpleBio.dnatm","page":"Primer Design","title":"SimpleBio.dnatm","text":"dnatm(dnaseq::String)\n\nCalculate the melting temperature of imput DNA sequence. The calculation is assume the annealing occurs under the standard conditions of 50 nM primer, 50 mM Na+, and pH 7.0.\n\n\n\n\n\n","category":"function"},{"location":"primer/#SimpleBio.step_gen","page":"Primer Design","title":"SimpleBio.step_gen","text":"step_gen(seq::String)\n\nGenerate a vector that contains stepwise increasing characters from the first character to the end.\n\n\n\n\n\n","category":"function"},{"location":"primer/#SimpleBio.step_from_end","page":"Primer Design","title":"SimpleBio.step_from_end","text":"step_from_end(seq::String)\n\nThe reverse version of step_gen()\n\n\n\n\n\n","category":"function"},{"location":"primer/#SimpleBio.multidnatm","page":"Primer Design","title":"SimpleBio.multidnatm","text":"multidnatm(dnas_array::Vector)\n\nGenerate a dataframe that shows all the Tm of sequences in the input array.\n\n\n\n\n\n","category":"function"},{"location":"primer/#SimpleBio.fwd_primers","page":"Primer Design","title":"SimpleBio.fwd_primers","text":"fwd_primers(seq::String)\n\nAuto generation of foward primer candidates of an input sequence.\n\n\n\n\n\nfwd_primers(seq::String, tm_low::Int64, tm_high::Int64)\n\nAuto generation of foward primer candidates of an input sequence in setted Tm range.\n\n\n\n\n\n","category":"function"},{"location":"primer/#SimpleBio.rev_primers","page":"Primer Design","title":"SimpleBio.rev_primers","text":"rev_primers(seq::String)\n\nAuto generation of foward primer candidates of an input sequence.\n\n\n\n\n\nrev_primers(seq::String, tm_low::Int64, tm_high::Int64)\n\nAuto generation of foward primer candidates of an input sequence in setted Tm range.\n\n\n\n\n\n","category":"function"},{"location":"#SimpleBio.jl","page":"SimpleBio","title":"SimpleBio.jl","text":"","category":"section"},{"location":"","page":"SimpleBio","title":"SimpleBio","text":"(Image: release) (Image: license) (Image: build) (Image: CodeCov) (Image: docs) (Image: docs) (Image: DOI)","category":"page"},{"location":"","page":"SimpleBio","title":"SimpleBio","text":"Welcome to SimpleBio.jl!","category":"page"},{"location":"","page":"SimpleBio","title":"SimpleBio","text":"Work with biological sequences as simple as a text way!","category":"page"},{"location":"#Features","page":"SimpleBio","title":"Features","text":"","category":"section"},{"location":"#.-Sequence-identification-for-DNA-or-RNA-by-isdna,-isrna-functions.","page":"SimpleBio","title":"1. Sequence identification for DNA or RNA by isdna, isrna functions.","text":"","category":"section"},{"location":"#.-Work-with-nucleotides:","page":"SimpleBio","title":"2. Work with nucleotides:","text":"","category":"section"},{"location":"","page":"SimpleBio","title":"SimpleBio","text":"Function transcript (xtranscript) for transcription of DNA to RNA (xtranscript for DNA with nucleobase not ATCG.)\nReverse-compliment sequence using DNArc and RNArc functions.\nRetro-Transcription of RNA to DNA using RTanscript function.\nFunction RandSeq for generating DNA, RNA or amino acid chains.\nUse the tranlatedna function for translation of dna sequence (also can enter RNA sequence if needed) to amino acid sequence.","category":"page"},{"location":"#.-Sequence-alignment-with-nalign-(DNA-or-RNA)-and-aalign-(for-amino-acid-sequences).","page":"SimpleBio","title":"3. Sequence alignment with nalign (DNA or RNA) and aalign (for amino acid sequences).","text":"","category":"section"},{"location":"#.-Outputs-are-text-String-that-can-be-used-in-the-further-processing!","page":"SimpleBio","title":"4. Outputs are text String that can be used in the further processing!","text":"","category":"section"},{"location":"#Install","page":"SimpleBio","title":"Install","text":"","category":"section"},{"location":"","page":"SimpleBio","title":"SimpleBio","text":"To install SimpleBio, type ']' in REPL to enter Pkg mode and type:","category":"page"},{"location":"","page":"SimpleBio","title":"SimpleBio","text":"add SimpleBio","category":"page"},{"location":"#Usage","page":"SimpleBio","title":"Usage","text":"","category":"section"},{"location":"","page":"SimpleBio","title":"SimpleBio","text":"To use SimpleBio, simply type in:","category":"page"},{"location":"","page":"SimpleBio","title":"SimpleBio","text":"using SimpleBio","category":"page"},{"location":"ali/","page":"Sequence Alignment","title":"Sequence Alignment","text":"CurrentModule = SimpleBio\nDocTestSetup = quote\n    using SimpleBio\nend","category":"page"},{"location":"ali/#Sequence-Alignment","page":"Sequence Alignment","title":"Sequence Alignment","text":"","category":"section"},{"location":"ali/#Align-the-DNA,-RNA-and-amino-acid-sequences","page":"Sequence Alignment","title":"Align the DNA, RNA and amino acid sequences","text":"","category":"section"},{"location":"ali/","page":"Sequence Alignment","title":"Sequence Alignment","text":"nalign\naalign","category":"page"},{"location":"ali/#SimpleBio.nalign","page":"Sequence Alignment","title":"SimpleBio.nalign","text":"nalign(type::Int64, s1::String, s2::String)\n\nAlignment of nucleotides (DNA or RNA) by default gapopen = -5, gapextend = -1 (see BioAlignments.jl for details.). parameter for type of sequence     1 => DNA     2 => RNA\n\n\n\n\n\nnalign(type::Int64, s1::String, s2::String, g_o::Int64, g_e::Int64)\n\nAlignment of nucleotides (DNA or RNA) with type-in gapopen and gapextend values.  (see BioAlignments.jl for details.)\n\n\n\n\n\n","category":"function"},{"location":"ali/#SimpleBio.aalign","page":"Sequence Alignment","title":"SimpleBio.aalign","text":"aalign(s1::String, s2::String)\n\nAlignment of amino acid by default gapopen = -5, gapextend = -1  (see BioAlignments.jl for details.)\n\n\n\n\n\naalign(s1::String, s2::String, g_o::Int64, g_e::Int64)\n\nAlignment of nucleotides (DNA or RNA) with type-in gapopen and gapextend values.  (see BioAlignments.jl for details.)\n\n\n\n\n\n","category":"function"},{"location":"nts/","page":"Work with Nucleotides","title":"Work with Nucleotides","text":"CurrentModule = SimpleBio\nDocTestSetup = quote\n    using SimpleBio\nend","category":"page"},{"location":"nts/#Work-with-nucleotides","page":"Work with Nucleotides","title":"Work with nucleotides","text":"","category":"section"},{"location":"nts/","page":"Work with Nucleotides","title":"Work with Nucleotides","text":"transcript\nxtranscript\nDNArc\nRNArc\nRTranscript\nRandSeq\nisdna\nisrna\ntranslatedna\nrand_dna_rna","category":"page"},{"location":"nts/#SimpleBio.transcript","page":"Work with Nucleotides","title":"SimpleBio.transcript","text":"transcript(x::String)\n\ntranscription of the imput DNA equence. This function only allow DNA sequence with A, T, C, G bases. If there are other kind of base in the DNA sequence, consider using xtranscript(x::String)\n\n\n\n\n\n","category":"function"},{"location":"nts/#SimpleBio.xtranscript","page":"Work with Nucleotides","title":"SimpleBio.xtranscript","text":"xtranscript(x::String)\n\ntranscription of the imput DNA equence. This function allow DNA sequences with bases not included in A, T, C, G.\n\n\n\n\n\n","category":"function"},{"location":"nts/#SimpleBio.DNArc","page":"Work with Nucleotides","title":"SimpleBio.DNArc","text":"DNArc(x::String)\n\nThis is the function for generating a reverse-compliment sequence to the entered sequence in string type.\n\n\n\n\n\n","category":"function"},{"location":"nts/#SimpleBio.RNArc","page":"Work with Nucleotides","title":"SimpleBio.RNArc","text":"RNArc(x::String)\n\nThis is the function for generating a reverse-compliment sequence to the entered sequence in string type.\n\n\n\n\n\n","category":"function"},{"location":"nts/#SimpleBio.RTranscript","page":"Work with Nucleotides","title":"SimpleBio.RTranscript","text":"RTranscript(x::String)\n\nGenerate the retro transcription of the entered sequence in string type.\n\n\n\n\n\n","category":"function"},{"location":"nts/#SimpleBio.RandSeq","page":"Work with Nucleotides","title":"SimpleBio.RandSeq","text":"RandSeq(type::String, length::Int64)\n\nGenerate random sequence of select type (RNA, DNA or aminoacid). Acceptable parameters of type:  DNA: [\"DNA\", \"DNAs\", \"DNA nucleotides\", \"DNA chain\", \"dna\", \"dnas\", \"dna nucleotides\", \"dna chain\"] RNA: [\"RNA\", \"RNAs\", \"RNA nucleotides\", \"RNA chain\", \"rna\", \"rnas\", \"rna nucleotides\", \"rna chain\"] amino acid: [\"AminoAcid\", \"aminoacid\", \"amino acid\", \"peptide\", \"aa\", \"AA\"]\n\n\n\n\n\nRandSeq(type::String, length::Int64, num::Int64)\n\nGenerate inputted numbers of random sequence of select type (RNA, DNA or aminoacid). Acceptable parameters of type:  DNA: [\"DNA\", \"DNAs\", \"DNA nucleotides\", \"DNA chain\", \"dna\", \"dnas\", \"dna nucleotides\", \"dna chain\"] RNA: [\"RNA\", \"RNAs\", \"RNA nucleotides\", \"RNA chain\", \"rna\", \"rnas\", \"rna nucleotides\", \"rna chain\"] amino acid: [\"AminoAcid\", \"aminoacid\", \"amino acid\", \"peptide\", \"aa\", \"AA\"]\n\n\n\n\n\nRandSeq(operator::Int64, type::String, length::Int64, num::Int64)\n\nGenerate inputted numbers of random sequence of select type (RNA, DNA or aminoacid). operator = 1 to get array returned. Acceptable parameters of type:  DNA: [\"DNA\", \"DNAs\", \"DNA nucleotides\", \"DNA chain\"] RNA: [\"RNA\", \"RNAs\", \"RNA nucleotides\", \"RNA chain\"] amino acid: [\"AminoAcid\", \"aminoacid\", \"amino acid\", \"peptide\", \"aa\", \"AA\"]\n\n\n\n\n\n","category":"function"},{"location":"nts/#SimpleBio.isdna","page":"Work with Nucleotides","title":"SimpleBio.isdna","text":"isdna(x::String)\n\nIdentify is a sequence is DNA sequence in string type.\n\n\n\n\n\n","category":"function"},{"location":"nts/#SimpleBio.isrna","page":"Work with Nucleotides","title":"SimpleBio.isrna","text":"isrna(x::String)\n\nIdentify is a sequence is RNA sequence in string type.\n\n\n\n\n\n","category":"function"},{"location":"nts/#SimpleBio.translatedna","page":"Work with Nucleotides","title":"SimpleBio.translatedna","text":"translatedna(x::String)\n\nReturn the translation result of the input DNA or RNA sequence in 3-letter amino acid codes.\n\n\n\n\n\ntranslatedna(operator::Int64, x::String)\n\nReturn the translation result of the input DNA or RNA sequence in 1-letter amino acid codes while operator enter =1.\n\n\n\n\n\n","category":"function"},{"location":"nts/#SimpleBio.rand_dna_rna","page":"Work with Nucleotides","title":"SimpleBio.rand_dna_rna","text":"rand_dna_rna(length::Int64, number::Int64)\n\nGenerate a dataframe of DNA sequences and transcripted RNAs with typed in length and number.\n\n\n\n\n\n","category":"function"}]
}
